function-images/tests/chained-function-serving/producer/producer.go

```diff
@@ -0,0 +1,136 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+
+	"google.golang.org/grpc"
+
+	pb_client "tests/chained-functions-serving/proto"
+
+	pb "github.com/ease-lab/vhive/examples/protobuf/helloworld"
+)
+
+type producerServer struct {
+	producerClient pb_client.ProducerConsumerClient
+	pb.UnimplementedGreeterServer
+}
+
+type producerServerLite struct {
+	pb.UnimplementedGreeterServer
+}
+
+func (ps *producerServer) SayHello(ctx context.Context, req *pb.HelloRequest) (*pb.HelloReply, error) {
+	ack, err := ps.producerClient.ConsumeString(context.Background(), &pb_client.ConsumeStringRequest{Value: "1"})
+	if err != nil {
+		log.Fatalf("[producer] client error in string consumption: %s", err)
+	}
+	log.Printf("[producer] (single) Ack: %v\n", ack.Value)
+	return &pb.HelloReply{Message: "Success"}, err
+}
+
+func main() {
+	lite := flag.Bool("lite", false, "Lite version of producer which does not require consumer")
+	flagAddress := flag.String("addr", "consumer.default.192.168.1.240.sslip.io", "Server IP address")
+	flagClientPort := flag.Int("pc", 80, "Client Port")
+	flagServerPort := flag.Int("ps", 3031, "Server Port")
+	flag.Parse()
+
+	//get address (from env or flag)
+	address, ok := os.LookupEnv("ADDR")
+	if !ok {
+		address = *flagAddress
+	}
+	var clientPort int
+	clientPortStr, ok := os.LookupEnv("PORT_CLIENT")
+	if !ok {
+		clientPort = *flagClientPort
+	} else {
+		var err error
+		clientPort, err = strconv.Atoi(clientPortStr)
+		if err != nil {
+			log.Fatalf("[producer] PORT_CLIENT env variable is not int: %v", err)
+		}
+	}
+	//get client port (from env of flag)
+	var serverPort int
+	serverPortStr, ok := os.LookupEnv("PORT")
+	if !ok {
+		serverPort = *flagServerPort
+	} else {
+		var err error
+		serverPort, err = strconv.Atoi(serverPortStr)
+		if err != nil {
+			log.Fatalf("[producer] PORT env variable is not int: %v", err)
+		}
+	}
+
+	//set up log file
+	file, err := os.OpenFile("log/logs.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
I think if docker-compose allows to see the logs in the terminal, we don't need to log to a file